Allow suppressing update check without suppressing error messages

Ruff has frequent releases, which is great, but it also means that the “new version available” message frequently adds noise to the output. Another source of noise is the “No pyproject.toml found. Falling back to default configuration…” message. A linter that prints expected noise can drown out unexpected errors, especially when it’s run from a script alongside many other linters.

```
No pyproject.toml found.
Falling back to default configuration...
Found 1 error(s).
test.py:1:5: W292 No newline at end of file

A new version of ruff is available: v0.0.82 -> v0.0.83
Run to update: pip3 install --upgrade ruff
```

Currently the only ways to suppress the update check are

* at compile time, by disabling the `update-informer` feature—not an option for those installing binaries;
* at runtime, with the `-q`/`--quiet` option—not an option for those who want to see the actual diagnostics about the input code.

There’s no way to suppress the pyproject.toml message.

Can we have an option to print *nothing* unless there are diagnostics about the input code? Perhaps this should be what `-q` means, and the existing `-q` should be `-qq` or `-s`/`--silent`?

--- a/src/cli.rs
+++ b/src/cli.rs
@@ -21,11 +21,14 @@ pub struct Cli {
     #[arg(long)]
     pub config: Option<PathBuf>,
     /// Enable verbose logging.
-    #[arg(short, long)]
+    #[arg(short, long, group = "verbosity")]
     pub verbose: bool,
-    /// Disable all logging (but still exit with status code "1" upon detecting errors).
-    #[arg(short, long)]
+    /// Only log errors.
+    #[arg(short, long, group = "verbosity")]
     pub quiet: bool,
+    /// Disable all logging (but still exit with status code "1" upon detecting errors).
+    #[arg(short, long, group = "verbosity")]
+    pub silent: bool,
     /// Exit with status code "0", even upon detecting errors.
     #[arg(short, long)]
     pub exit_zero: bool,

--- a/src/main.rs
+++ b/src/main.rs
@@ -220,7 +220,8 @@ fn autoformat(files: &[PathBuf], settings: &Settings) -> Result<usize> {
 }
 
 fn inner_main() -> Result<ExitCode> {
-    let cli = Cli::parse();
+    let mut cli = Cli::parse();
+    cli.quiet |= cli.silent;
 
     set_up_logging(cli.verbose)?;
 
@@ -342,7 +343,7 @@ fn inner_main() -> Result<ExitCode> {
         tell_user!("Starting linter in watch mode...\n");
 
         let messages = run_once(&cli.files, &settings, !cli.no_cache, false)?;
-        if !cli.quiet {
+        if !cli.silent {
             printer.write_continuously(&messages)?;
         }
 
@@ -362,7 +363,7 @@ fn inner_main() -> Result<ExitCode> {
                             tell_user!("File change detected...\n");
 
                             let messages = run_once(&cli.files, &settings, !cli.no_cache, false)?;
-                            if !cli.quiet {
+                            if !cli.silent {
                                 printer.write_continuously(&messages)?;
                             }
                         }
@@ -388,13 +389,13 @@ fn inner_main() -> Result<ExitCode> {
                 let path = Path::new(&filename);
                 (
                     run_once_stdin(&settings, path, cli.fix)?,
-                    !cli.quiet && !cli.fix,
+                    !cli.silent && !cli.fix,
                     false,
                 )
             } else {
                 (
                     run_once(&cli.files, &settings, !cli.no_cache, cli.fix)?,
-                    !cli.quiet,
+                    !cli.silent,
                     !cli.quiet,
                 )
             };

